File: content/courses/parallel-computing-introduction/codes/omp_parfor.f90
program omp_hello
use omp_lib
implicit none

   integer :: i, N, nthreads
   integer :: tid

   N=40
   nthreads=4

   !$omp parallel do num_threads(nthreads)
   do i=1,N
      tid=omp_get_thread_num()
      write(*,'(a,i3,a,i3)') "Thread ",tid," runs i= ",i
   enddo
   !$omp end parallel do

end program

File: content/courses/parallel-computing-introduction/codes/omp_parfor.c
#include <stdio.h>
#include <omp.h>

int main() {

    int N, nthreads;
    int tid;

    N=40;
    nthreads=4;

#pragma omp parallel for num_threads(nthreads)
    for (int i=0; i<N; i++) {
        tid=omp_get_thread_num();
        printf("Thread %d runs i= %d\n",tid, i);
    }
}
